### ISSUE TITLE
Compatible to Samsung Model H-Series?

### ISSUE THREAD

**Issue by HMside** [CONNCTED/homebridge-samsungtv#4]
Which TV-Series are compatible to your Homebridge Plugin?

**Comment by jonferreira** [comment]
same question, I mean is this compatible with J series?

**Comment by nowkillkennys** [comment]
same  here ive got a ue48h6500

**Comment by kkapitan** [comment]
Hi, guys i've made a plugin that should work with H and J series. It is still in progress (turning on does not work right now) but all of the other commands should be possible. You can find it [here](https://github.com/kkapitan/homebridge-homesung) just follow the README to set things up. Feel free to test it and leave your comments :) Hope it will solve our issues with that models.

**Comment by jonferreira** [comment]
hi @kkapitan 

I have a JU7500 so will give it a try soon, even though I don't use homekit any more (all my automation is done via node-red now)

Don't think you will ever be able to turn it On unfortunately, as far as I know the Ethernet port is disabled then the TV is On so no way to wake up the TV other than via IR

**Comment by kkapitan** [comment]
Hi @jonferreira ! Thanks, I am now working to make things more stable as some people with J series reported that they are unable to do the initial pairing.  As for turning on - I am yet to test the wake on lan feature, but on the device specs side I've noticed, what you said, that this feature is not enabled in H series. But my plan is to fallback to hdmi-cec solution which should work (given you are running homebridge on raspberry pi connected via hdmi to your TV).

**Comment by jonferreira** [comment]
ah yes that should work if you get the PI connected to the TV via HDMI, nice workaround!

**Comment by kkapitan** [comment]
It seems that J series issues were fixed, so you can test the plugin if you want! :)

**Comment by jonferreira** [comment]
ah cool nice work @kkapitan 

so does that means you now have the "wake up" working then? presumably with hdmi-cec?

**Comment by kkapitan** [comment]
The hdmi-cec thing is still in progress (I'm waiting for my RPI to come so I can test it, and unfortunately my development machine does not support sending CEC commands). 

The issue I spoke about was the inability for some J-series models to pair with the plugin, but this was now tested by those experiencing the issue and it looks like it's gone :) 

Have a great day!

**Comment by jonferreira** [comment]
ah great, thanks for letting me know.

I haven't used homebridge for a long time now and honestly don't see the need for it now *but* I'm wondering if I can port your code as a node-red plugin...

**Comment by kkapitan** [comment]
I see! My plan for the future was to extract all the logic responsible for the communication with the device to another npm package, so people would be able to build their own integrations without the necessity of using  homebridge.

I think you can do it even now, if you are impatient :) The project was structured so the homebridge glue code lies in the `/platform` folder and the connection and pairing code lies in `/src`. The file `homesung.js` is an entry point that can perform all the operations like requesting/confirming pin, connecting to the device and sending keys. 

Maybe this will help you :)

**Comment by jonferreira** [comment]
Yes, thanks will take a look at it when I have some free time :)
